lib/rules/no-trailing-spaces.js

```diff
@@ -87,10 +101,14 @@ module.exports = {
                 const re = new RegExp(NONBLANK),
                     skipMatch = new RegExp(SKIP_BLANK),
                     lines = sourceCode.lines,
-                    linebreaks = sourceCode.getText().match(astUtils.createGlobalLinebreakMatcher());
+                    linebreaks = sourceCode.getText().match(/\r\n|\r|\n|\u2028|\u2029/g),
+                    comments = sourceCode.getAllComments(),
```

Comment:
Note that in v4.0.0 this will now include shebangs (which I'm assuming we don't want to include in this)